### PR TITLE
feat: enhance clean cmd capability

### LINKED ISSUE #1425: enhance capabilities of clean command by specifying intent explicitly
**Description**

Besides the talk on the Slack channel: https://sigstore.slack.com/archives/C01PZKDL4DP/p1643905351672249

The existing clean command would be adjusted to delete more things, like “anything associated with image X”, sigs, attachments, attestations, etc.

And this new optional flag for clean would give users a chance to pare that down to just one kind of thing, like “just attachments” for example.

> Thanks to @luhring explained it very well

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>

#### Summary

To let people specify which tag type they want to delete from an image, .sig or .att or both.

#### Ticket Link

Fixes #1425 

#### Release Note

```release-note
feat: enhance clean cmd capability
```

### CODE REVIEW

File: cmd/cosign/cli/clean.go [hunk 1]
```diff
@@ -29,41 +29,58 @@ import (
...
+	attRef, err := ociremote.AttestationTag(ref, ociremote.WithRemoteOptions(remoteOpts...))
 	if err != nil {
 		return err
 	}
 
+	var cleanTags []name.Tag
+	switch cleanType {
+	case "signature":
+		cleanTags = []name.Tag{sigRef}
+	case "attestation":
```
Comment (dlorenc) [inline cmd/cosign/cli/clean.go:70]: Should we include SBOMs too?

File: test/e2e_test.go [hunk 2]
```diff
@@ -167,8 +167,8 @@ func TestSignVerifyClean(t *testing.T) {
 	must(download.SignatureCmd(ctx, options.RegistryOptions{}, imgName), t)
 
 	// Now clean signature from the given image
-	must(cli.CleanCmd(ctx, options.RegistryOptions{}, imgName), t)
-
+	must(cli.CleanCmd(ctx, options.RegistryOptions{}, "all", imgName), t)
+	1
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply test/e2e_test.go]: thank you 🤦

**Comment by dlorenc** [comment]
Looks like we currently get an error if one of the types is not present:
image

We should probably allow that and continue.

File: cmd/cosign/cli/clean.go [hunk 3]
```diff
@@ -29,41 +29,65 @@ import (
...
+		cleanTags = []name.Tag{attRef}
+	case "all":
+		cleanTags = []name.Tag{sigRef, attRef, sbomRef}
+	}
+
+	for _, t := range cleanTags {
+		fmt.Fprintf(os.Stderr, "Removing %s from %s\n", t.String(), imageRef)
+
+		err = remote.Delete(t, remoteOpts...)
+		if err != nil {
```
Comment (dlorenc) [inline cmd/cosign/cli/clean.go:87]: These should probably just be logged and continued instead of fatal.

**Review by dlorenc — APPROVED** [review]

**Pull request merged**